drivers/serial/uart_native_tty.c

```diff
@@ -48,171 +44,65 @@ struct native_tty_config {
 	struct uart_config uart_config;
 };
 
-struct baudrate_termios_pair {
-	int baudrate;
-	speed_t termios_baudrate;
-};
-
 /**
- * @brief Lookup table for mapping the baud rate to the macro understood by termios.
- */
-static const struct baudrate_termios_pair baudrate_lut[] = {
-	{1200, B1200},       {1800, B1800},       {2400, B2400},       {4800, B4800},
-	{9600, B9600},       {19200, B19200},     {38400, B38400},     {57600, B57600},
-	{115200, B115200},   {230400, B230400},   {460800, B460800},   {500000, B500000},
-	{576000, B576000},   {921600, B921600},   {1000000, B1000000}, {1152000, B1152000},
-	{1500000, B1500000}, {2000000, B2000000}, {2500000, B2500000}, {3000000, B3000000},
-	{3500000, B3500000}, {4000000, B4000000},
-};
-
-/**
- * @brief Set given termios to defaults appropriate for communicating with serial port devices.
+ * @brief Convert from uart_config to native_tty_bottom_cfg eqvivalent struct
  *
- * @param ter
+ * @param bottom_cfg
+ * @param cfg
  */
-static inline void native_tty_termios_defaults_set(struct termios *ter)
+static void native_tty_conv_to_bottom_cfg(struct native_tty_bottom_cfg *bottom_cfg,
+					  const struct uart_config *cfg)
 {
-	/* Set terminal in "serial" mode:
-	 *  - Not canonical (no line input)
-	 *  - No signal generation from Ctr+{C|Z..}
-	 *  - No echoing
-	 */
-	ter->c_lflag &= ~(ICANON | ISIG | ECHO);
-
-	/* No special interpretation of output bytes.
-	 * No conversion of newline to carriage return/line feed.
-	 */
-	ter->c_oflag &= ~(OPOST | ONLCR);
-
-	/* No software flow control. */
-	ter->c_iflag &= ~(IXON | IXOFF | IXANY);
-
-	/* No blocking, return immediately with what is available. */
-	ter->c_cc[VMIN] = 0;
-	ter->c_cc[VTIME] = 0;
-
-	/* No special handling of bytes on receive. */
-	ter->c_iflag &= ~(IGNBRK | BRKINT | PARMRK | ISTRIP | INLCR | IGNCR | ICRNL);
-
-	/* - Enable reading data and ignore control lines */
-	ter->c_cflag |= CREAD | CLOCAL;
-}
-
-/**
- * @brief Set the baud rate speed in the termios structure
- *
- * @param ter
- * @param baudrate
- *
- * @retval 0		If successful,
- * @retval -ENOTSUP	If requested baud rate is not supported.
- */
-static inline int native_tty_baud_speed_set(struct termios *ter, int baudrate)
-{
-	for (int i = 0; i < ARRAY_SIZE(baudrate_lut); i++) {
-		if (baudrate_lut[i].baudrate == baudrate) {
-			cfsetospeed(ter, baudrate_lut[i].termios_baudrate);
-			cfsetispeed(ter, baudrate_lut[i].termios_baudrate);
-			return 0;
-		}
+	bottom_cfg->baudrate = cfg->baudrate;
+
+	switch (cfg->parity) {
+	case UART_CFG_PARITY_NONE: {
+		bottom_cfg->parity = NTB_PARITY_NONE;
+	} break;
+	case UART_CFG_PARITY_ODD: {
+		bottom_cfg->parity = NTB_PARITY_ODD;
+	} break;
+	case UART_CFG_PARITY_EVEN: {
+		bottom_cfg->parity = NTB_PARITY_EVEN;
+	} break;
+	default: {
+		bottom_cfg->parity = NTB_PARITY_UNSUPPORTED;
```

Comment:
Would it nicer to have this function return already a -ENOTSUP instead of setting it to NTB_PARITY_UNSUPPORTED, 
and have the caller check for a return != 0 and if so propagate the returned error up?
(same for the other not supported cases)

That way these conditions would be already caught gracefully here, and you can save the code in the bottom that checks for them (or have the bottom code just error).
Say `native_tty_data_bits_set()` would error in the default instead of returning -ENOTSUP, and `native_tty_configure_bottom()` would not need to check for the return of `native_tty_data_bits_set()` (and others).